File: apps/web/src/types/file.ts
import type { ApiPagination } from './api'

export type FileItem = {
  id: string
  tenant_id: string
  parent_file_id: string | null
  upload_id: string | null
  original_name: string
  normalized_name?: string | null
  extension?: string | null
  detected_mime_type?: string | null
  detected_file_type?: string | null
  storage_path: string
  size_bytes: number
  sha256_hash?: string | null
  depth: number
  is_archive: boolean
  is_password_protected: boolean
  processing_status: string
  created_by?: string | null
  created_at: string
  updated_at: string
}

export type ParsedRecord = {
  id: string
  tenant_id: string
  file_id: string
  job_id: string
  record_type: string | null
  record_number: number | null
  line_number: number | null
  chunk_number: number | null
  start_line: number | null
  end_line: number | null
  content_text: string | null
  structured_data: Record<string, unknown> | null
  extracted_entities: Record<string, unknown> | null
  event_timestamp: string | null
  created_at: string
}

export type ListFilesParams = {
  tenant_id?: string
  tenantId?: string
  status?: string
  extension?: string
  detected_file_type?: string
  detectedFileType?: string
  page?: number
  page_size?: number
  pageSize?: number
}

export type FileListResponse = ApiPagination & {
  files: FileItem[]
}

export type FileChildrenResponse = ApiPagination & {
  file_id: string
  children: FileItem[]
}

export type FileTreeNode = {
  file: FileItem
  children: FileTreeNode[]
}

export type FileRecordsResponse = ApiPagination & {
  file_id: string
  records: ParsedRecord[]
}

export type UploadInitiateRequest = {
  file_name: string
  content_type: string
  size_bytes: number
  password_provided: boolean
}

export type UploadInitiateResponse = {
  upload_id: string
  object_key: string
  upload_url: string
  expires_in_seconds: number
}

export type UploadCompleteRequest = {
  upload_id: string
}

export type UploadCompleteResponse = {
  upload_id: string
  file_id: string
  job_id: string
  status: string
}

export type SubmitFilePasswordRequest = {
  password: string
}

export type SubmitFilePasswordResponse = {
  file_id: string
  job_id: string
  status: string
}

export type UploadPagination = ApiPagination

File: apps/api/internal/models/file.go
package models

import (
	"time"
)

type File struct {
	ID                 string          `json:"id"`
	TenantID           string          `json:"tenant_id"`
	ParentFileID       *string         `json:"parent_file_id,omitempty"`
	UploadID           *string         `json:"upload_id,omitempty"`
	OriginalName       string          `json:"original_name"`
	NormalizedName     *string         `json:"normalized_name,omitempty"`
	Extension          *string         `json:"extension,omitempty"`
	DetectedMimeType   *string         `json:"detected_mime_type,omitempty"`
	DetectedFileType   *string         `json:"detected_file_type,omitempty"`
	StoragePath        string          `json:"storage_path"`
	SizeBytes          int64           `json:"size_bytes"`
	Sha256Hash         *string         `json:"sha256_hash,omitempty"`
	Depth              int             `json:"depth"`
	IsArchive          bool            `json:"is_archive"`
	IsPasswordProtected bool           `json:"is_password_protected"`
	ProcessingStatus   string          `json:"processing_status"`
	CreatedBy          *string         `json:"created_by,omitempty"`
	CreatedAt          time.Time       `json:"created_at"`
	UpdatedAt          time.Time       `json:"updated_at"`
}
